plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -219,6 +220,40 @@ public static Table getIcebergTableWithMetadata(
         return new BaseTable(operations, quotedTableName(table), TRINO_METRICS_REPORTER);
     }
 
+    public static List<IcebergColumnHandle> getProjectedColumns(Schema schema, TypeManager typeManager)
+    {
+        ImmutableList.Builder<IcebergColumnHandle> projectedColumns = ImmutableList.builder();
+        StructType schemaAsStruct = schema.asStruct();
+        Map<Integer, NestedField> indexById = TypeUtil.indexById(schemaAsStruct);
+        Map<Integer, Integer> indexParents = TypeUtil.indexParents(schemaAsStruct);
+
+        for (Map.Entry<Integer, NestedField> entry : indexById.entrySet()) {
```

Comment:
AFAIU we do call `getPath` fo all the `indexById` entries.
We could instead have a utility method which returns a `Map<Integer, List<Integer>` `<id, path>` upfront this mapping.

Reply:
It would be also beneficial to have a unit test in `TestIcebergUtil` for this method.